File: iosApp/iosApp/Presentation/Main/MainAction.swift
//
//  MainAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation

enum MainAction {
    case enter(preferredRoute: String? = nil)
    case refreshNavigation(preferredRoute: String? = nil)
    case selectTab(route: String)
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/MainAction.kt
package com.hhp227.concafe.presentation.main

sealed interface MainAction {
    data class CheckAppUpdate(
        val storePlatform: String,
        val storeId: String,
        val currentVersion: String
    ) : MainAction
    data class Enter(val preferredRoute: String? = null) : MainAction
    data class RefreshNavigation(val preferredRoute: String? = null) : MainAction
    data class SelectTab(val route: String) : MainAction
}
